src/test/java/org/apache/commons/imaging/common/ImageBuilderTest.java

```diff
@@ -0,0 +1,146 @@
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.commons.imaging.common;
+
+import java.awt.image.BufferedImage;
+import java.awt.image.ColorModel;
+import java.awt.image.RasterFormatException;
+
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * Provides unit tests for the ImageBuilder class.
+ */
+public class ImageBuilderTest {
+
+
+    /**
+     * Test of bad dimensions in constructor
+     */
+    @Test
+    public void testConstructorBounds() {
+        executeBadConstructor(0, 10);
+        executeBadConstructor(10, 0);
+    }
+
+
+    /**
+     * Test of bad bounds in sub-image
+     */
+    @Test
+    public void testBoundsCheck() {
+
+        ImageBuilder imageBuilder = new ImageBuilder(100, 100, false );
+
+        executeBadBounds(imageBuilder, -1,  0, 50, 50);
+        executeBadBounds(imageBuilder,  0, -1, 50, 50);
+        executeBadBounds(imageBuilder,  0,  0,  0, 50);
+        executeBadBounds(imageBuilder,  0,  0, 50, 0);
+        executeBadBounds(imageBuilder, 90,  0, 50, 50);
+        executeBadBounds(imageBuilder,  0, 90, 50, 50);
+    }
+
+    /**
+     * Test whether sub-image is consistent with source
+     */
+    @Test
+    public void testSubimageAccess() {
+        ImageBuilder imageBuilder = new ImageBuilder(100, 100, false );
+        populate(imageBuilder);
+        BufferedImage bImage = imageBuilder.getSubimage(25, 25, 25, 25);
+        int w = bImage.getWidth();
+        int h = bImage.getHeight();
+        assertEquals(w, 25, "Width of subimage does not match");
+        assertEquals(h, 25, "Height of subimage does not match");
+
+        for(int x=25; x<50; x++){
+            for(int y=25; y<50; y++){
+                int k = bImage.getRGB(x-25, y-25);
+                int rgb = imageBuilder.getRGB(x, y);
+                assertEquals(k, rgb, "Invalid buffered image subpixel at "+x+", "+y);
+            }
+        }
+
+        ImageBuilder testBuilder = imageBuilder.getSubset(25, 25, 25, 25);
+        for(int x=25; x<50; x++){
+            for(int y=25; y<50; y++){
+                int k = testBuilder.getRGB(x-25, y-25);
+                int rgb = imageBuilder.getRGB(x, y);
+                assertEquals(k, rgb, "Invalid image builder subpixel at "+x+", "+y);
+            }
+        }
+    }
```

Comment:
These tests :point_up:  are for the other change with subset right? Just to confirm :+1:

Reply:
Yes.  I'm trying to bring the coverage back to 100 percent for ImageBuilder